quasar-core/src/main/java/co/paralleluniverse/common/util/ExtendedStackTrace.java

```diff
@@ -109,11 +113,13 @@ public void visitLineNumber(int line, Label start) {
                                         minLine = line;
                                     if (line > maxLine)
                                         maxLine = line;
+                                    if (lineNumber == line)
+                                        exactMatch.set(desc);
                                 }
 
                                 @Override
                                 public void visitEnd() {
-                                    if (minLine <= este.getLineNumber() && maxLine >= este.getLineNumber())
+                                    if (minLine <= lineNumber && maxLine >= lineNumber && descriptor.get() == null)
                                         descriptor.set(desc);
```

Comment:
Isn't this racy? Perhaps use `descriptor.compareAndSet(null, desc)` instead? Otherwise why are we using an `AtomicReference` in the first place?

Reply:
What if you get two Fibers in two or more threads both having their stack checked at the same time ?
I agree it's a bit sad if they should interfere with each other but the original author used AtomicReference for a reason.

Reply:
I think we are all good because the `AtomicReference` is a local variable. So the second thread will have it's own independent `AtomicReference` to work with.

Reply:
I'll make the change to use `compareAndSet`